src/ui/wxWidgets/PasswordSafeSearch.cpp

```diff
@@ -361,72 +455,73 @@ void PasswordSafeSearch::ReCreateSearchBar(void)
+
+  auto panel = new wxPanel(m_parentFrame, wxID_ANY);
+  auto panelSizer = new wxBoxSizer(wxHORIZONTAL);
   panel->SetSizer(panelSizer);
-  m_toolbar = new wxToolBar(panel, wxID_ANY, wxDefaultPosition, wxDefaultSize, wxBORDER_NONE | wxTB_BOTTOM | wxTB_HORIZONTAL,  wxT("SearchBar"));
+
+  m_toolbar = new wxToolBar(panel, wxID_ANY, wxDefaultPosition, wxDefaultSize, wxTB_BOTTOM | wxTB_DEFAULT_STYLE,  wxT("SearchBar"));
   panelSizer->Add(m_toolbar, wxSizerFlags().Proportion(1).Expand());
 
+  // Close Button
   m_toolbar->AddTool(ID_FIND_CLOSE, wxEmptyString, wxBitmap(findclose_xpm), wxNullBitmap, wxITEM_NORMAL, _("Close SearchBar"));
-  wxSize srchCtrlSize(m_parentFrame->GetSize().GetWidth()/3, wxDefaultSize.GetHeight());
-  wxSearchCtrl* srchCtrl = new wxSearchCtrl(m_toolbar, ID_FIND_EDITBOX, wxEmptyString, wxDefaultPosition, srchCtrlSize, wxTE_PROCESS_ENTER);
+
+  // Search Textfield
+  auto srchCtrl = new wxSearchCtrl(m_toolbar, ID_FIND_EDITBOX, wxEmptyString, wxDefaultPosition, CalculateSearchWidth(), wxTE_PROCESS_ENTER);
   srchCtrl->ShowCancelButton(true);
   srchCtrl->ShowSearchButton(true);
-
   m_toolbar->AddControl(srchCtrl);
+
+  // Search Tools
   m_toolbar->AddTool(ID_FIND_NEXT, wxEmptyString, wxBitmap(find_xpm), wxBitmap(find_disabled_xpm), wxITEM_NORMAL, _("Find Next"));
   m_toolbar->AddCheckTool(ID_FIND_IGNORE_CASE, wxEmptyString, wxBitmap(findcase_i_xpm), wxBitmap(findcase_s_xpm), _("Case Insensitive Search"));
   m_toolbar->AddTool(ID_FIND_ADVANCED_OPTIONS, wxEmptyString, wxBitmap(findadvanced_xpm), wxNullBitmap, wxITEM_CHECK, _("Advanced Find Options"));
   m_toolbar->AddTool(ID_FIND_CREATE_REPORT, wxEmptyString, wxBitmap(findreport_xpm), wxNullBitmap, wxITEM_NORMAL, _("Create report of previous Find search"));
   m_toolbar->AddTool(ID_FIND_CLEAR, wxEmptyString, wxBitmap(findclear_xpm), wxNullBitmap, wxITEM_NORMAL, _("Clear Find"));
-  m_toolbar->AddControl(new wxStaticText(m_toolbar, ID_FIND_STATUS_AREA, wxEmptyString, wxDefaultPosition, srchCtrlSize.Scale(3,1), wxTE_READONLY));
+  m_toolbar->AddSeparator();
+
+  // Status Area
+  m_toolbar->AddControl(new wxStaticText(m_toolbar, ID_FIND_STATUS_AREA, wxEmptyString, wxDefaultPosition, wxDefaultSize, wxALIGN_LEFT|wxST_ELLIPSIZE_END));
 
   RefreshButtons();
 
-  if (!m_toolbar->Realize())
+  if (!m_toolbar->Realize()) {
     wxMessageBox(_("Could not create Search Bar"), _("Password Safe"));
+  }
 
-  wxSizer* origSizer = m_parentFrame->GetSizer();
+  auto origSizer = m_parentFrame->GetSizer();
   wxASSERT(origSizer);
   wxASSERT(origSizer->IsKindOf(wxBoxSizer(wxVERTICAL).GetClassInfo()));
   wxASSERT(((wxBoxSizer*)origSizer)->GetOrientation() == wxVERTICAL);
   origSizer->Add(panel, 0, wxEXPAND);
   origSizer->Layout();
-  if (!m_toolbar->Show(true) && !m_toolbar->IsShownOnScreen())
-    wxMessageBox(_("Could not display searchbar"));
 
-  //This gross hack is the only way I could think of to get ESC keystrokes from the text ctrl user is typing into
```

Comment:
What happened to this? Can't see where this has been implemented or what replaced it in the new code.

Reply:
This block of code to catch the ESC key event has been replaced by code at line 517.

Line 517:
`m_toolbar->Bind(wxEVT_CHAR_HOOK,                     &PasswordSafeSearch::OnChar,                  this);`